src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -42,17 +45,29 @@ class HttpKernel implements HttpKernelInterface, TerminableInterface
     /**
      * Constructor.
      *
-     * @param EventDispatcherInterface    $dispatcher   An EventDispatcherInterface instance
-     * @param ControllerResolverInterface $resolver     A ControllerResolverInterface instance
-     * @param RequestStack                $requestStack A stack for master/sub requests
+     * @param EventDispatcherInterface    $dispatcher              An EventDispatcherInterface instance
+     * @param ControllerResolverInterface $resolver                A ControllerResolverInterface instance
+     * @param RequestStack                $requestStack            A stack for master/sub requests
+     * @param ArgumentResolverManager     $argumentResolverManager A manager to resolve the controller arguments
      *
      * @api
      */
-    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null)
+    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null, ArgumentResolverManager $argumentResolverManager = null)
     {
         $this->dispatcher = $dispatcher;
         $this->resolver = $resolver;
         $this->requestStack = $requestStack ?: new RequestStack();
+
+        if (null === $argumentResolverManager) {
+            $argumentResolverManager = new ArgumentResolverManager(array(
+                new RequestArgumentResolver(),
+                new RequestAttributesArgumentResolver(),
+            ));
+        }
+
+        if (method_exists($resolver, 'setArgumentResolverManager')) {
```

Comment:
The `ControllerResolver#setArgumentResolverManager()` method is tagged as `@internal`, which means you shouldn't use it in your own code. It only exists, so we can call it here in the HttpKernel.

This is to be BC. In 3.0, `ControllerResolver#setArgumentResolverManager()` will be removed and the complete argument resolving part will be called by HTTP kernel directly (ControllerResolver will only resolve the controller then). So if you want to add custom argument resolvers, do it by customizing HttpKernel instead of ControllerResolver.